compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -707,8 +710,11 @@ ConfigurationAdminImpl::AddingService(
                 "obtained from the BundleContext");
     return nullptr;
   }
-  // Ensure there's a Configuration for this PID if one doesn't exist already.
+  // Lock the configurations repository so no configuration objects can be
+  // added or removed while AddingService is processing the new service.
+  std::lock_guard<std::mutex> lk{ configurationsMutex };
 
+  // Ensure there's a Configuration for this PID if one doesn't exist already.
```

Comment:
I'm not sure we should be creating a configuration object here. There are two ways that valid configuration objects are created. A configuration object can appear in a manifest.json file (in that case it is added by AddConfigurations and an Updated notification will be sent at that time.) The other way is in user code when a user gets a configuration object and then updates it. This code here is creating a blank configuration object which is meaningless. Unless the configuration object is updated by a user or updated by AddConfigurations, the configuration object has no value. It doesn't hurt to add it here. It just might not be necessary.